Dynatrace/dynatrace-operator
# Fix usage of TrustedCAs in the Dynakube

# Description

Fixes https://github.com/Dynatrace/dynatrace-operator/issues/903

Currently our dynakube controller hits a nil pointer, if there is a custom certificate added via `trustedCAs` field in the dynakube. This happened because, the wrong constructor was used to initialise the DynakubeController. Furthermore, after fixing that problem I discovered that we pass the wrong path to docker, as it expects the path to the certificates directory and not the path to the actual file, which we create for the trustedCAs.

One minor thing was, that the CSI Driver now also has to access the configmap, which was passed via the Dynakube, so the SA needs correct permissions to GET configmaps.

## How can this be tested?

1. Create a configmap containing the certificates you want to trust e.g. `kubectl create configmap -n dynatrace dynakubecas --from-file=./certs` (beware the content of the cm has to be called certs)
2. Afterwards create a dynakube and pass the configmap name within the Dynakubes trustedCAs field
3. check if everything works as expected and the API Calls are successful


## Checklist
